pkg/wireguard/agent/node_handler.go

```diff
@@ -0,0 +1,68 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package agent
+
+import (
+	datapath "github.com/cilium/cilium/pkg/datapath/types"
+	"github.com/cilium/cilium/pkg/logging/logfields"
+	nodeTypes "github.com/cilium/cilium/pkg/node/types"
+)
+
+// NodeAdd is called when a node is discovered for the first time.
+func (a *Agent) NodeAdd(newNode nodeTypes.Node) error {
+	if newNode.IsLocal() {
+		return nil
+	}
+
+	if newNode.WireguardPubKey != "" {
+		newIP4 := newNode.GetNodeIP(false)
+		newIP6 := newNode.GetNodeIP(true)
+
+		if err := a.UpdatePeer(newNode.Fullname(), newNode.WireguardPubKey, newIP4, newIP6); err != nil {
+			log.WithError(err).
+				WithField(logfields.NodeName, newNode.Fullname()).
+				Warning("Failed to update wireguard configuration for peer")
+		}
+	}
+
+	return nil
```

Comment:
Consider unifying the NodeAdd and NodeUpdate body. They both don't care about the oldNode, so maybe just having a `nodeUpsert` helper would do the trick

Reply:
yep good point; done